File: scripts/common_js/GUOHUB_CLI_CONFIG.ts
import { readFileSync, existsSync } from "node:fs";
import { parse } from "smol-toml";

function loadConfig(): Record<string, unknown> {
  const root = process.env.GUOHUB_ROOT;
  const path = root ? `${root}/config.toml` : new URL("../../config.toml", import.meta.url).pathname;
  if (!existsSync(path)) return { keyring_namespace: "GUOHUB_CLI" };
  return parse(readFileSync(path, "utf-8")) as Record<string, unknown>;
}

export const NAME_SPACE = String(loadConfig().keyring_namespace ?? "GUOHUB_CLI");

File: scripts/common_py/GUOHUB_CLI_CONFIG.py
import os
import tomllib
from pathlib import Path


def _load_config():
    root = os.environ.get("GUOHUB_ROOT")
    if root:
        path = Path(root) / "config.toml"
    else:
        path = Path(__file__).resolve().parent.parent.parent / "config.toml"
    if not path.exists():
        return {"keyring_namespace": "GUOHUB_CLI"}
    with open(path, "rb") as f:
        return tomllib.load(f)


NAME_SPACE = _load_config().get("keyring_namespace", "GUOHUB_CLI")
